rdmo/projects/tests/test_view_project_visibility.py

```diff
@@ -0,0 +1,188 @@
+
+
+def test_detail_site_forbidden(db, client):
+    client.login(username='user', password='user')
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.set(Site.objects.filter(id=2))
+    project.visibility.groups.clear()
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 403
+
+
+def test_detail_group(db, client):
+    client.login(username='user', password='user')
+
+    group = Group.objects.create(name='test')
+    user = User.objects.get(username='user')
+    user.groups.add(group)
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.clear()
+    project.visibility.groups.add(group)
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 200
+
+
+def test_detail_multiple_groups(db, client):
+    client.login(username='user', password='user')
+
+    group = Group.objects.create(name='test')
+    user = User.objects.get(username='user')
+    user.groups.add(group)
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.clear()
+    project.visibility.groups.set([
+        group, Group.objects.create(name='test2')
+    ])
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 200
+
+
+def test_detail_group_forbidden(db, client):
+    client.login(username='user', password='user')
+
+    group = Group.objects.create(name='test')
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.clear()
+    project.visibility.groups.set([group])
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 403
+
+
+def test_detail_site_group(db, client):
+    client.login(username='user', password='user')
+
+    group = Group.objects.create(name='test')
+    user = User.objects.get(username='user')
+    user.groups.add(group)
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.set(Site.objects.filter(id=1))
+    project.visibility.groups.set([group])
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 200
+
+
+def test_detail_site_group_forbidden(db, client):
+    client.login(username='user', password='user')
+
+    group = Group.objects.create(name='test')
+    user = User.objects.get(username='user')
+    user.groups.add(group)
+
+    project = Project.objects.get(id=project_id)
+    project.visibility.sites.set(Site.objects.filter(id=2))
+    project.visibility.groups.set([group])
+
+    url = reverse('project', args=[project_id])
+    response = client.get(url)
+    assert response.status_code == 403
+
+
+# @pytest.mark.parametrize('username,password', users)
+# @pytest.mark.parametrize('project_id', projects)
+# def test_project_update_visibility_get(db, client, username, password, project_id):
```

Comment:
what about these tests?

Reply:
Ah, I need to remove them, they moved to `test_view_project_update_visibility.py`.

Reply:
done.